packages/utils/typescript/lib/generators/utils.js

```diff
@@ -0,0 +1,222 @@
+    'placeholder.ts',
+    '',
+    ts.ScriptTarget.ESNext,
+    true,
+    ts.ScriptKind.TS
+  );
+
+  const printer = ts.createPrinter({ omitTrailingSemicolon: true });
+
+  return printer.printList(ts.ListFormat.MultiLine, nodeArray, sourceFile);
+};
+
+/**
+ * Save the given string representation of TS nodes in a file
+ * If the given directory doesn't exist, it'll be created automatically
+ *
+ * @param {string} dir
+ * @param {string} file
+ * @param {string} content
+ *
+ * @return {Promise<string>} The path of the created file
+ */
+const saveDefinitionToFileSystem = async (dir, file, content) => {
+  const filepath = path.join(dir, file);
+
+  fse.ensureDirSync(dir);
+  await fse.writeFile(filepath, content);
+
+  return filepath;
+};
+
+/**
+ * Format the given definitions.
+ * Uses the existing config if one is defined in the project.
+ *
+ * @param {string} content
+ * @returns {Promise<string>}
+ */
+const format = async (content) => {
+  const configFile = await prettier.resolveConfigFile();
+  const config = configFile
+    ? await prettier.resolveConfig(configFile)
+    : // Default config
+      {
+        singleQuote: true,
+        useTabs: false,
+        tabWidth: 2,
+      };
+
+  Object.assign(config, { parser: 'typescript' });
+
+  return prettier.format(content, config);
+};
+
+/**
+ * Generate the import declaration for local directory
+ *
+ * @returns {ts.ImportDeclaration}
+ */
+const generateLocalDirectoryImportDefinition = () => {
+  return factory.createImportDeclaration(
+    undefined,
+    undefined,
+    factory.createStringLiteral('.'),
+    undefined
+  );
+};
+
+/**
+ * Generate the extension block for a shared component from strapi/strapi
+ *
+ * @param {string} registry The registry to extend
+ * @param {Array<{ uid: string; definition: ts.TypeNode }>} definitions
+ * @returns {ts.ModuleDeclaration}
+ */
+const generateSharedExtensionDefinition = (registry, definitions) => {
+  const properties = definitions.map(({ uid, definition }) =>
+    factory.createPropertySignature(
+      undefined,
+      factory.createStringLiteral(uid, true),
+      undefined,
+      factory.createTypeReferenceNode(factory.createIdentifier(definition.name.escapedText))
+    )
+  );
+
+  return factory.createModuleDeclaration(
+    [factory.createModifier(ts.SyntaxKind.DeclareKeyword)],
+    factory.createStringLiteral('@strapi/strapi', true),
+    factory.createModuleBlock([
+      factory.createModuleDeclaration(
+        [factory.createModifier(ts.SyntaxKind.ExportKeyword)],
+        factory.createIdentifier('Shared'),
+        factory.createModuleBlock([
+          factory.createInterfaceDeclaration(
+            [factory.createModifier(ts.SyntaxKind.ExportKeyword)],
+            factory.createIdentifier(registry),
+            undefined,
+            undefined,
+            properties
+          ),
+        ])
+      ),
+    ]),
+    ts.NodeFlags.ExportContext
+  );
+};
+
+const createLogger = (options = {}) => {
```

Comment:
Do we have to create our own logger? Can we lean on winston or something else that's already in Strapi?

Reply:
I would rather have a common CLI logger, but it was not in the scope of this task.
We can change that later if we decide to unify everything.

In any case, I'm not sure we should re-use Strapi's logger :thinking: 
What do you think?